addon/ng2/blueprints/component/index.js

```diff
@@ -16,22 +19,59 @@ module.exports = {
   },
 
   locals: function (options) {
+    //TODO: pull value from config
+    this.styleExt = 'css';
+    
     return {
       dynamicPath: this.dynamicPath.dir.replace(this.dynamicPath.appRoot, ''),
-      flat: options.flat
+      flat: options.flat,
+      styleExt: this.styleExt,
+      isLazyRoute: !!options.isLazyRoute,
+      isAppComponent: !!options.isAppComponent
     };
   },
+  
+  files: function() {
+    var fileList = getFiles.call(this);
+    
+    if (this.options.flat) {
+      fileList = fileList.filter(p => p.indexOf('index.ts') <= 0);
+    }
 
+    return fileList;
+  },
+  
   fileMapTokens: function (options) {
     // Return custom template variables here.
     return {
       __path__: () => {
         var dir = this.dynamicPath.dir;
         if (!options.locals.flat) {
           dir += path.sep + options.dasherizedModuleName;
+          
+          if (options.locals.isLazyRoute) {
+            var dirParts = dir.split(path.sep);
+            dirParts[dirParts.length - 1] = `+${dirParts[dirParts.length - 1]}`;
+            dir = dirParts.join(path.sep);
+          }
         }
+        this.appDir = dir.replace(`src${path.sep}client${path.sep}`, '');
         return dir;
+      },
+      __styleext__: () => {
+        return options.locals.styleExt;
       }
     };
+  },
+  
+  afterInstall: function(options) {
+    if (!options.flat) {
+      var filePath = path.join('src', 'client', 'system-config.ts');
+      return this.insertIntoFile(
+        filePath,
+        `  '${this.appDir}',`,
+        { before: '  //barrel registration' }
```

Comment:
Thinking about it, a token comment would work too but I'd rather see something like

```
/** @cli-barrel */
```

And then look for `@cli-barrel.**/` in case the user wants to refactor this file. Might work too, just make sure we use a token that identifiable.